docs/topics/addons.rst

```diff
@@ -0,0 +1,189 @@
+   be able to load the class of the fallback component from a special setting
+   (e.g. ``MY_FALLBACK_DOWNLOAD_HANDLER``), create an instance of it and use
+   it.
+2. The add-ons that include these components should read the current value of
+   the default setting (e.g. ``DOWNLOAD_HANDLERS``) in their
+   ``update_settings()`` methods, save that value into the fallback setting
+   (``MY_FALLBACK_DOWNLOAD_HANDLER`` mentioned earlier) and set the default
+   setting to the component provided by the add-on (e.g.
+   ``MyDownloadHandler``). If the fallback setting is already set by the user,
+   they shouldn't change it.
+3. This way, if there are several add-ons that want to modify the same setting,
+   all of them will fallback to the component from the previous one and then to
+   the Scrapy default. The order of that depends on the priority order in the
+   ``ADDONS`` setting.
+
+
+Add-on examples
+===============
+
+Set some basic configuration:
+
+.. code-block:: python
+
+    class MyAddon:
+        def update_settings(self, settings):
+            settings["ITEM_PIPELINES"]["path.to.mypipeline"] = 200
+            settings.set("DNSCACHE_ENABLED", True, "addon")
+
+Check dependencies:
+
+.. code-block:: python
+
+    class MyAddon:
+        def update_settings(self, settings):
+            try:
+                import boto
+            except ImportError:
+                raise RuntimeError("MyAddon requires the boto library")
+            ...
+
+Access the crawler instance:
+
+.. code-block:: python
+
+    class MyAddon:
+        def __init__(self, crawler) -> None:
+            super().__init__()
+            self.crawler = crawler
+
+        @classmethod
+        def from_crawler(cls, crawler):
+            return cls(crawler)
+
+        def update_settings(self, settings):
+            ...
+
+Use a fallback component:
+
+.. code-block:: python
+
+    from scrapy.core.downloader.handlers.http import HTTPDownloadHandler
+
+
+    FALLBACK_SETTING = "MY_FALLBACK_DOWNLOAD_HANDLER"
+
+
+    class MyHandler:
+        lazy = False
+
+        def __init__(self, settings, crawler):
+            dhcls = load_object(settings.get(FALLBACK_SETTING))
+            self._fallback_handler = create_instance(
+                dhcls,
+                settings=None,
+                crawler=crawler,
+            )
+
+        def download_request(self, request, spider):
+            if request.meta.get("my_params"):
+                # handle the request
+                ...
+            else:
+                return self._fallback_handler.download_request(request, spider)
+
+
+    class MyAddon:
+        def update_settings(self, settings):
+            if not settings.get(FALLBACK_SETTING):
+                settings.set(
+                    FALLBACK_SETTING,
+                    settings.getwithbase("DOWNLOAD_HANDLERS")["https"],
+                    "addon",
+                )
+            settings["DOWNLOAD_HANDLERS"]["https"] = "path.to.MyHandler"
```

Comment:
can you use MyHandler here directly? Or would you consider it an anti-pattern? Not sure myself :)